### DISCUSSION TITLE
Segment length keeps decreasing for long recordings?

### DISCUSSION THREAD

**Discussion by s-h-a-d-o-w** [openai/whisper#1695] [Q&A]
I just ran a university lecture through whisper and when scrolling through it, it stood out to me that the average segment length keeps shrinking as time goes on.

Now, for this particular purpose, it probably makes sense to merge everything into one line and then manually split it into blocks anyway. Still... in general, this is probably not great UX. For example subtitles would definitely have to be post-processed. If that can be avoided, it would obviously be great.

I'm surprised that this hasn't come up yet. At least I haven't found anything...

[liessmann.txt](https://github.com/openai/whisper/files/12835236/liessmann.txt)

**Comment by whicks1** [comment]
I see this in most of my files. We post process the vtt and split the outputs after attempting to calculate a good split point between word and punctuation boundaries and a guesstimated reasonable max line length and then do some maths on the durations. Vtt library is your friend for that sorta thing. Timestamps aren’t perfect but it’s close enough for us.  If you want to stick to the project, you can look at —help for the max length and related arguments to give a whirl as well.
